src/spyglass/lfp/v1/analytic_signal.py

```diff
@@ -0,0 +1,141 @@
+    """
+
+    def insert_default(self):
+        """Insert the default artifact parameters with an appropriate parameter dict."""
+        analytic_signal_params = dict()
+        analytic_signal_params["analytic_method_name"] = "hilbert_transform"
+        analytic_signal_params[
+            "other_params"
+        ] = None  # Hilbert transform doesn't contain other parameters; alternative methods may contain additional params.
+        self.insert1(["default", analytic_signal_params], skip_duplicates=True)
+
+    def insert1(self, key, skip_duplicates):
+        if key[1]["analytic_method_name"] == "hilbert_transform":
+            super().insert1(key, skip_duplicates=skip_duplicates)
+        else:
+            raise KeyError(
+                f"analytic_method_name '{key[1]['analytic_method_name']}' is not currently supported."
+            )
+
+
+@schema
+class AnalyticSignalSelection(dj.Manual):
+    definition = """
+    # Table that combines the LFP data, and parameters for generating analytic signals
+    -> LFPBand
+    -> IntervalList
+    -> LFPElectrodeGroup
+    -> AnalyticSignalParameters
+    ---
+    """
+
+
+@schema
+class AnalyticSignal(dj.Computed):
+    definition = """
+    -> AnalyticSignalSelection
+    ---
+    -> AnalysisNwbfile # the name of the nwb file with the analytic sinal & time stamps data
+    analytic_signal_object_id: varchar(40)  # the NWB object ID for loading this object from the file
+    """
+
+    def make(self, key):
+        # First, get the filtered lfp data
+        filtered_band = (LFPBand() & key).fetch_nwb()[0]["filtered_data"]
+
+        # Next, select data using defined electrode id(s) and defined valid interval.
+        electrode_list = (LFPElectrodeGroup.LFPElectrode() & key).fetch("electrode_id")
+        electrode_index = np.isin(filtered_band.electrodes.data[:], electrode_list)
+        start, end = (IntervalList() & key).fetch1("valid_times")[0]
+        timestamps_selected = filtered_band.timestamps[
+            np.logical_and(
+                filtered_band.timestamps >= start, filtered_band.timestamps <= end
+            )
+        ]
+        filtered_band_selected = filtered_band.data[
+            np.logical_and(
+                filtered_band.timestamps >= start, filtered_band.timestamps <= end
+            )
+        ][:, electrode_index]
+
+        # Get analytic signal with the specified method (we're only using Hilbert transform for now, and the algorithm will raise errors if other methods are defined by the user)
+        filtered_analytic = np.empty(
+            shape=(filtered_band_selected.shape[0], 0), dtype=np.complex_
+        )
+        params = (AnalyticSignalParameters() & key).fetch1("analytic_signal_params")
+        for lfp_electrode_index in range(filtered_band_selected.shape[1]):
+            if params["analytic_method_name"] == "hilbert_transform":
+                analytic_signal_one_channel = hilbert(
```

Comment:
I think `scipy.signal.hilbert` loads the entire input signal onto memory, and thus does not scale well for long chunks of recordings. I would suggest using `ghostipy` to do this, e.g. the [`analytic_signal`](https://github.com/kemerelab/ghostipy/blob/b34ec0954dbfa7fd638279e96ad58cfd04fee988/ghostipy/dsp/analytic.py#L31) function.

Reply:
@khl02007 I believe people only use their reference channels (1-2 channels at a time) for computing theta phase, so the memory load might not be an issue even if we use data from a full day. Let me know if it's absolutely necessary to replace scipy by ghostipy. I'm doing a speed test on my scripts now with the full-day data and will report back.

Reply:
I did a quick test with one run session of data & two electrodes, and it took < 30s to populate the analytic signal table. I believe a full day of ~15sessions probably won't take more than 1 hour and shouldn't raise a memory concern. Does that sound reasonable to you? @khl02007

Reply:
@dpeg22 @xlsun79 @edeno I thought the analytic signal is used for analyses other than estimating theta phase, like detecting SWR. Is there no situation where we would pass a long recording from more than a single electrode? On the other hand, if this is only for computing theta phase from a single electrode and it doesn't take very long to compute, do we really need to save this to NWB file or have this as a separate pipeline? Isn't it faster to do `scipy.signal.hilbert` to a pre-extracted theta band from `LFPBand` and then do `np.angle` or `np.abs(...)**2` to get phase or power?

Reply:
It is never used in SWR detection, but it is a common enough analysis that we do so perhaps it makes sense to have it standardized in our pipelines. We do have a general question of how many things should be stored vs. computed on the fly but we haven't come to any decisions on standards for that so I think it is okay for this to be part of the LFP pipeline for now.